runtime/src/main/java/org/corfudb/util/NetworkUtils.java

```diff
@@ -18,6 +23,25 @@
 @Slf4j
 public class NetworkUtils {
 
+
+    /**
+     * Get all interfaces, including virtual interfaces.
+     * @return A list of all interfaces
+     * @throws SocketException
+     */
+    private static List<NetworkInterface> getAllInterfaces() throws SocketException {
+        List<NetworkInterface> allInterfaces = new ArrayList<>(50);
+        Stack<NetworkInterface> stack = new Stack<>();
```

Comment:
Using Deque is preferred to Stack (which also provides all the functionalities of a stack : https://docs.oracle.com/javase/7/docs/api/java/util/Stack.html)